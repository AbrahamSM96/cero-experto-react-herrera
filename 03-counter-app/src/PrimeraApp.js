import React from "react";
//Functional Components
const PrimeraApp = () => {
    return (
        <>
            <h1>Hola perros en functional component</h1>
        </>
    );
};

export default PrimeraApp;
