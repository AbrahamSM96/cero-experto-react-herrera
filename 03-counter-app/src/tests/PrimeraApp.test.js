import { render } from "@testing-library/react";
import '@testing-library/jest-dom'
import PrimeraApp from "../PrimeraApp";
import { shallow } from 'enzyme'

describe('Pruebas en <PrimeraApp></PrimeraApp>', () => {
    // test('Debe de mostrar el mensaje "I`m Abraham Serrano"', () => {
    //     const saludo = "I`m abraham serrano"
    //     const {getByText} = render(<PrimeraApp saludos={saludo} />)
        
    //     expect(getByText(saludo  )).toBeInTheDocument();

    // });
    test('debe de mostrar <PrimeraApp/> correctamente', () => {
        const saludo = "Hola, soy abrahaaaam"
        const wrapper = shallow(<PrimeraApp saludos={saludo} />)
        
        expect(wrapper).toMatchSnapshot();

    });

    test('debe de mostrar el subtitulo enviado por props', () => {
        
        const saludo = "Hola, soy abrahaaaam"
        const subtitulo = "Hola soy un subtitulo"
        const wrapper = shallow(<PrimeraApp saludos={saludo} subtitulos={subtitulo} />);
        // find actuara como un queryselector
        const textoParrafo = wrapper.find('p').text()
        // probamos que el subtitulo sea igual al del componente
        expect( textoParrafo).toBe(subtitulo);
    });
})