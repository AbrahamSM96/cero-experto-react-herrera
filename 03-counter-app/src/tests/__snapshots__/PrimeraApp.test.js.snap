// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimeraApp></PrimeraApp> debe de mostrar <PrimeraApp/> correctamente 1`] = `
<Fragment>
  <h1>
    Hola, soy abrahaaaam
    !!
  </h1>
  <p>
    Soy un default subtitulo
  </p>
</Fragment>
`;
